src/librustc/session/config.rs

```diff
@@ -663,6 +663,8 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
           "print the result of the translation item collection pass"),
     mir_opt_level: Option<usize> = (None, parse_opt_uint,
           "set the MIR optimization level (0-3)"),
+    dump_mir: Option<String> = (None, parse_opt_string,
```

Comment:
@nagisa 

> Is --unpretty={mir[=function],mir-cfg[=function]} somehow unsatisfactory/not enough?

It is indeed unsatisfactory for two reasons:

1. It dumps the IR only after all the work is done, rather than dumping the state in between every pass and during processing. 
2. It offers no filters on the state that is dumped. i.e., usually I'm debugging a problem in some particular function or is some particular pass, and I'd like to be able to just dump the MIR for that.
3. In general `--unpretty` does not fit well with our stabilization scheme. I do not see us *ever* stabilizing MIR dumps -- or at least not for a very long time -- which means that `-Z` is a more appropriate place for the option to live. This has the added benefit that we don't have to write `-Z unstable-options` when using it, which is annoying to do. (If you find yourself writing `-Z unstable-options`, and the flag feels like something internal, it should probably just *be* a `-Z` flag.)